packages/trader/src/App/Components/Elements/VideoPlayer/volume-control.tsx

```diff
@@ -0,0 +1,175 @@
+    };
+
+    const mouseUpHandler = (e: MouseEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+
+        if (volume_bar_filled_ref.current) setIsAnimated(true);
+        is_dragging.current = false;
+
+        // Hide volume bar if user finished changing volume outside of the container
+        if (!(e.target as HTMLElement).closest('.player__volume-bar__wrapper')) setShowVolume(false);
+    };
+
+    const mouseDownHandler = (e: React.MouseEvent<HTMLSpanElement>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const point = volume_dot_ref.current;
+        setShiftY((point?.getBoundingClientRect().bottom ?? 0) - e.clientY);
+
+        is_dragging.current = true;
+    };
+
+    const onRewind = (e: React.MouseEvent<HTMLDivElement>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if ((e.target as HTMLElement).className === 'player__volume-dot') return;
+        if (!volume_bar_filled_ref.current) return;
+
+        setIsAnimated(false);
+        const new_height = calculateNewHight(e);
+        volume_bar_filled_ref.current.style.setProperty('height', `${new_height}%`);
+
+        onVolumeChange(new_height / 100);
+        checkVolumeControl(new_height);
+    };
+
+    React.useEffect(() => {
+        document.addEventListener('mousemove', mouseMoveHandler);
+        document.addEventListener('mouseup', mouseUpHandler);
+
+        return () => {
+            document.removeEventListener('mousemove', mouseMoveHandler);
+            document.removeEventListener('mouseup', mouseUpHandler);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <div
+            className='player__volume__wrapper'
+            onMouseOver={is_mobile ? undefined : () => setShowVolume(true)}
+            onMouseLeave={is_mobile || is_dragging.current ? undefined : () => setShowVolume(false)}
+        >
+            <button className='player__controls__button' onClick={buttonClickHandler}>
+                <Icon
+                    icon={is_muted ? 'IcSoundOff' : 'IcSoundOn'}
+                    custom_color='var(--text-colored-background)'
+                    height={18}
+                    width={20}
+                />
+            </button>
+            <CSSTransition
+                in={show_volume}
+                timeout={100}
+                classNames={{
+                    enter: 'player__volume-bar__wrapper--enter',
+                    enterDone: 'player__volume-bar__wrapper--enter-done',
+                    exit: 'player__volume-bar__wrapper--exit',
+                }}
+                unmountOnExit
+            >
+                <div className='player__volume-bar__wrapper'>
+                    <div
+                        className='player__volume-bar'
+                        onClick={onRewind}
+                        onKeyDown={undefined}
```

Comment:
SonarcCloud still complaining, let's add `role='button'` instead of empty `onKeyDown`s. please check all similar occurencies  🙏🏻